database/src/main/postgres/runs/V1.5.12__write_additional_data.sql

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+CREATE OR REPLACE FUNCTION runs.write_additional_data(
+    IN  i_partitioning      JSONB,
+    IN  i_additional_data   HSTORE,
+    IN  i_by_user           TEXT,
+    OUT status              INTEGER,
+    OUT status_text         TEXT,
+    OUT id_additional_data  BIGINT
+) RETURNS record AS
+$$
+-------------------------------------------------------------------------------
+--
+-- Function: runs.write_additional_data(3)
+--      Adds the additional data for the input partitioning. If additional data of a given name already
+--      exists for such partitioning, the value is updated and the old value is moved to the
+--      additional data history table.
+--
+-- Parameters:
+--      i_partitioning      - partitioning to add the additional data for
+--      i_additional_data   - sets of key/value pairs representing name and values of the additional data
+--      i_by_user           - user behind the change
+--
+-- Returns:
+--      status              - Status code
+--      status_text         - Status text
+--      id_additional_data  - id of the data added
+--
+-- Status codes:
+--      11                  - Additional data have been added
```

Comment:
The application is unfortunately blind to what happens at the database server as these statuses are not propagated back to the application. I see this as a big problem that should be addressed as soon as possible. What if we wanted to act upon the resulting status code and status text?

Reply:
We already talked about this a few times. I agree, this feels definitely like a missing feature, I just created a ticket for it: https://github.com/AbsaOSS/fa-db/issues/115

To the best of my knowledge, no application needed it, neither it might need it in the near future, but I would like to have it there, it absolutely makes sense